assets/src/stories-editor/helpers/index.js

```diff
@@ -1354,10 +1368,8 @@ export const maybeInitializeAnimations = () => {
 			const page = getBlockRootClientId( block.clientId );
 
 			if ( page ) {
-				const { ampAnimationType, ampAnimationAfter, ampAnimationDuration, ampAnimationDelay } = block.attributes;
-				const predecessor = allBlocks.find( ( b ) => b.attributes.anchor === ampAnimationAfter );
-
-				addAnimation( page, block.clientId, predecessor ? predecessor.clientId : undefined );
```

Comment:
@swissspidy FYI: the issue seemed to come from the animation of predecessor not being initialized before adding the animation -- this made `isValidAnimationPredecessor` always fail. Well, not always, depending on the order of the blocks.

Haven't tested thoroughly yet but seems to work now.